OpenSim/Common/TableSource.h

```diff
@@ -75,16 +84,33 @@ class TableSource_ : public Component {
     OpenSim_DECLARE_LIST_OUTPUT(column, ET, getColumnAtTime, 
                                 SimTK::Stage::Time);
 
-    TableSource_()                               = default;
+    TableSource_() {
+        constructProperties();
+    }
+
     TableSource_(const TableSource_&)            = default;
     TableSource_(TableSource_&&)                 = default;
     TableSource_& operator=(const TableSource_&) = default;
     TableSource_& operator=(TableSource_&&)      = default;
     
     /** Construct the TableSource_ by giving it a TimeSeriesTable_ to hold.   */
-    TableSource_(const Table& table) : 
-        _table{table} 
-    {}
+    TableSource_(const Table& table) :
+        _table{table} {
+        constructProperties();
+    }
+
+    /** Construct the TableSource_ from a file.
+
+    \param filename Name of the file.
+    \param tablename Name of the table in the file to populate the TableSource
+                     with. Ex. 'markers', 'forces'.
+
+    \throws KeyNotFound If table provided does not have column-labels.        */
+    TableSource_(const std::string& filename,
+                 const std::string& tablename) {
+        constructProperties();
```

Comment:
In many cases a file will contain only one table, but it is necessary to provide a tablename. Perhaps a default empty string should be provided and in the case that their are more tables throw if the specific table name was not provided.